bin/ocamllsp.ml

```diff
@@ -0,0 +1,133 @@
+open! Import
+module Pkg_dev_tool = Dune_rules.Pkg_dev_tool
+
+let ocamllsp_exe_path = Path.build @@ Pkg_dev_tool.exe_path Ocamllsp
+let ocamllsp_exe_name = Pkg_dev_tool.exe_name Ocamllsp
+let compiler_package_name = Package_name.of_string "ocaml"
+
+module Failure = struct
+  module Reason = struct
+    type t =
+      | Not_in_dune_project
+      | No_lock_dir
+      | No_ocaml_lockfile_in_lock_dir of { lock_dir_path : Path.Source.t }
+
+    let to_string = function
+      | Not_in_dune_project -> "you don't appear to be inside a dune project"
+      | No_lock_dir ->
+        sprintf
+          "the current dune project lacks a lockdir associated with the default context"
+      | No_ocaml_lockfile_in_lock_dir { lock_dir_path } ->
+        sprintf
+          "the lockdir %S lacks a lockfile for the %S package"
+          (Path.Source.to_string lock_dir_path)
+          (Package_name.to_string compiler_package_name)
+    ;;
+  end
+
+  (* A message explaining why dune can't install/run ocamllsp as a
+     dev-tool. *)
+  let message reason =
+    User_message.make
+      [ Pp.textf
+          "Unable to run %s as a dev-tool because %s."
+          ocamllsp_exe_name
+          (Reason.to_string reason)
+      ; Pp.nop
+      ; Pp.textf
+          "To run %s as a dev-tool, the following conditions mest be met:"
```

Comment:
```suggestion
          "To run %s as a dev-tool, the following conditions must be met:"
```

Reply:
Nobody is going to run `dune lsp` - it's always going to be started by your text editor. In a later PR I'm going to make it so that if the conditions necessary for ocamllsp to be installed as a dev tool are not met, that dune will try to fall back to a system or opam-managed instance of ocamllsp. This is necessary so that users can put `dune ocamllsp` in their editor configs but still be able to edit ocamlc code without a lockfile or even outside a dune project. Without this I don't expect many people to start using `dune ocamllsp`.